### ISSUE TITLE
Refactor validation

### ISSUE THREAD

**Issue by armintaenzertng** [spdx/tools-python#307]
We want a refactored validation using the new data model (available on https://github.com/spdx/tools-python/tree/refactor-python-tools). The main goal is to keep it completely separated from the parsing/writing layers.

Validation error messages could be instances of a class that tracks their context like this:
```
class ErrorMessage:
    element_type: SpdxElementType # an enum with values document, package, file, snippet, relationship, annotation, ...
    id: Optional[str] # not every type has an id or it might be missing
    validation_messages: List[str]
```
Validation can be performed by calling validation methods that return `List[ErrorMessage]`. Keeping them in separate classes allows for better unit testing by using mocks. The spdx version will be passed to validators that differ for different versions.
```
class DocumentValidator:
    spdx_version: str
    creation_info_validator = CreationInfoValidator()
    package_validator = PackageValidator(spdx_version)
    file_validator = FileValidator(spdx_version)
    ...

    def validate_document(self, document: Document) -> List[ErrorMessage]:
        error_messages = []
        error_messages.extend(creation_info_validator.validate_creation_info(document))
        error_messages.extend(package_validator.validate_packages(document.packages))
        ...
        return error_messages
``` 
```
class CreationInfoValidator()
    def validate_creation_info(self, document: Document) -> List[ErrorMessage]:
        error_messages = []
        spdx_id: Optional[str]

        ... # check and set the spdx_id
        
        if document.namespace is None:
            error_messages.append(ErrorMessage(SpdxElementType.Document, spdx_id, 'Missing required field "namespace."'))
```

**Comment by armintaenzertng** [comment]
Completion overview:

- [ ] actor
- [ ] annotation
- [ ] checksum
- [ ] external document ref
- [ ] extracted licensing info
- [ ] file
- [ ] license
- [ ] package
- [ ] relationship
- [ ] snippet

**Comment by armintaenzertng** [comment]
This has been implemented in 0.8.0
